src/Stack/Build/Execute.hs

```diff
@@ -1141,8 +1141,11 @@ singleBuild runInBase ac@ActionContext {..} ee@ExecuteEnv {..} task@Task {..} in
                             ("Warning: haddock not generating hyperlinked sources because 'HsColour' not\n" <>
                              "found on PATH (use 'stack install hscolour' to install).")
                         return ["--hyperlink-source" | hscolourExists]
+            let internalDocs = if boptsHaddockInternal bopts
```

Comment:
It's the `bopts` that aren't in scope and causing the Travis failures. From a superficial glance at the code above: are you possibly looking for the `eeBuildOpts`?